insolar/pulse/db.go

```diff
@@ -78,6 +86,29 @@ func (s *DB) Latest(ctx context.Context) (pulse insolar.Pulse, err error) {
 	return nd.Pulse, nil
 }
 
+// TruncateHead remove all records after lastPulse
+func (s *DB) TruncateHead(ctx context.Context, lastPulse insolar.PulseNumber) error {
+	it := s.db.NewIterator(pulseKey(math.MaxUint32), true)
+	defer it.Close()
+
+	for it.Next() {
+		key := newPulseKey(it.Key())
+		if insolar.PulseNumber(key) <= lastPulse {
+			if !insolar.PulseNumber(key).Equal(lastPulse) {
+				return errors.New("[ DB.TruncateHead ] No required pulse: " + lastPulse.String())
```

Comment:
We don't use special error formatting.
https://insolar.atlassian.net/wiki/spaces/DEV/pages/738131969/Go+code+style#Gocodestyle-6Errors
6.1